src/txmempool.h

```diff
@@ -542,7 +542,7 @@ class CTxMemPool
     template <typename Lambda>
     void forEachThenClear(const Lambda &f)
     {
-        WRITELOCK(cs);
+        AssertWriteLockHeld(cs);
```

Comment:
the purpose of the forEachThenClear function was to bury the lock handling.  mapTx is a public variable so if a set of operations are needed at a higher level, we can take the lock and iterate manually.

Reply:
done